packages/angular_devkit/build_angular/src/webpack/plugins/karma/karma.ts

```diff
@@ -167,46 +149,54 @@ const init: any = (config: any, emitter: any) => {
   webpackConfig.output.path = `/${KARMA_APPLICATION_PATH}/`;
   webpackConfig.output.publicPath = `/${KARMA_APPLICATION_PATH}/`;
 
-  let compiler;
-  try {
-    compiler = webpack(webpackConfig);
-  } catch (e) {
-    logger.error(e.stack || e)
-    if (e.details) {
-      logger.error(e.details)
+  const compiler = webpack(webpackConfig, (error, stats) => {
+    if (error) {
+      throw error;
     }
-    throw e;
-  }
 
-  function handler(callback?: () => void) {
-    isBlocked = true;
+    if (stats?.hasErrors()) {
+      // Only generate needed JSON stats and when needed.
+      const statsJson = stats?.toJson({
+        all: false,
+        children: true,
+        errors: true,
+        warnings: true,
+      });
 
-    if (typeof callback === 'function') {
-      callback();
+      logger.error(statsErrorsToString(statsJson, { colors: true }));
+
+      // Notify potential listeners of the compile error.
+      emitter.emit('compile_error', {
+        errors: statsJson.errors,
```

Comment:
`statsJson.errors` can be an object with a message property.  Does the `compile_error` event support that?

Reply:
Mapped to retain current behaviour.